pinot-common/src/main/java/org/apache/pinot/common/metrics/ValidationMetrics.java

```diff
@@ -239,6 +239,14 @@ public void cleanupSegmentCountGauge(final String resource) {
     removeGauge(resource, ValidationMetricName.SEGMENT_COUNT);
   }
 
+  public void updateTmpSegmentCountGauge(final String resource, final long tmpSegmentCount) {
```

Comment:
Why do we define it as a gauge? Should it be a meter instead?

Let's match the name of the method and the metric

Reply:
```  
public enum ValidationMetricName {
    MISSING_SEGMENT_COUNT("missingSegmentCount"),
    OFFLINE_SEGMENT_DELAY_HOURS("offlineSegmentDelayHours"),
    LAST_PUSH_TIME_DELAY_HOURS("lastPushTimeDelayHours"),
    TOTAL_DOCUMENT_COUNT("TotalDocumentCount"),
    NON_CONSUMING_PARTITION_COUNT("NonConsumingPartitionCount"),
    SEGMENT_COUNT("SegmentCount"),
    DELETED_TMP_SEGMENT_COUNT("DeletedTmpSegmentCount");
```
It seems that all count metrics are using Gauge

Reply:
Shall we use Count?

Reply:
```  
public static <T> PinotGauge<T> makeGauge(PinotMetricsRegistry registry, PinotMetricName name, PinotGauge<T> gauge) {
    return registry.newGauge(name, gauge);
  }

  public static PinotMeter makePinotMeter(PinotMetricsRegistry registry, PinotMetricName name, String eventType,
      TimeUnit unit) {
    return registry.newMeter(name, eventType, unit);
  }
```
Meter metrics would output .m1, .m5, .m15 I am not whether we need those statics. Existing count related metrics of ValidationMetrics also use gauge

Reply:
Gauge means refreshing the value, so that can be used to for total docs, total segments etc, but not for incremental values such as segments deleted in this round

Reply:
Appreciate for sharing the knowledge. I have replace gauge with meter and add removeTableMeter method into AbstractMetrics.

Reply:
@Jackie-Jiang  Could you have another review?